app/java/src/main/java/com/google/cloudclientapi/CensusController.java

```diff
@@ -16,28 +16,65 @@
 
 package com.google.cloudclientapi;
 
-
+import com.google.gson.Gson;
 import java.util.Arrays;
 import java.util.List;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 
 @Controller
 public class CensusController {
+
+  private static final String PROCESSED_DATA_BUCKET = System.getenv().get("PROCESSED_DATA_BUCKET");
   private static final Logger logger = LoggerFactory.getLogger(CensusController.class);
 
   @GetMapping("/")
-  public String main(Model model) {
-    logger.info("Received request (TODO: data)");
+  public String main(
+      @RequestParam(name = "fur", required = false, defaultValue = "Gray") String fur,
+      @RequestParam(name = "age", required = false, defaultValue = "Adult") String age,
+      @RequestParam(name = "location", required = false, defaultValue = "Above Ground")
+          String location,
+      Model model) {
+    logger.info("Request received for fur: {}, age: {}, location: {}", fur, age, location);
```

Comment:
nit: please consider using structure payload that is formatted to follow Google Cloud structured log best practices.

Reply:
Do you have a reference implementation for @NimJay to follow?

Reply:
I've very intentionally matched the logs (and even _some_ function names) in this Java app with the existing Python app.
I vote we consider adding structured logging out-of-scope for this pull-request — in the interest of time.
@glasnt feel free to re-open if you believe structured logging should be added in this PR itself.